client/api/src/notifications.rs

```diff
@@ -114,10 +156,101 @@ impl<Block: BlockT> Default for StorageNotifications<Block> {
 	}
 }
 
+struct PrintKeys<'a>(&'a Option<HashSet<StorageKey>>);
+impl<'a> std::fmt::Display for PrintKeys<'a> {
+	fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
+		if let Some(keys) = self.0.as_ref() {
+			write!(fmt, "[")?;
+			let mut is_first = true;
+			for key in keys {
+				if is_first {
+					is_first = false;
+				} else {
+					write!(fmt, ", ")?;
+				}
+				write!(fmt, "{}", HexDisplay::from(key))?;
+			}
+			write!(fmt, "]")
```

Comment:
```suggestion
			fmt.debug_list().entries(keys.iter().map(HexDisplay::from))
```

Reply:
This won't compile exactly as you wrote, but great idea! I completely forgot that this exists.

Reply:
Yeah sorry, github sadly doesn't provide a rust compiler :D But maybe something in that direction.